develop/tutorials/articles/200-integrating-with-front-end-frameworks/01-integrating-with-react.markdown

```diff
@@ -0,0 +1,201 @@
+proceed with a manual configuration. This tutorial shows how to use React in 
+your portlets, whether you're migrating an existing React project or building a 
+fresh project. See the 
+[npm React portlet template](/develop/reference/-/knowledge_base/7-1/npm-react-portlet-template) 
+reference docs for more information on the portlet's anatomy or the 
+[react npm portlet sample](/develop/reference/-/knowledge_base/7-1/react-npm-portlet) 
+for a React portlet example that you can test and deploy right now. Get started 
+by creating your OSGi module and configuring its metadata. 
+
+## Configuring Metadata [](id=configuring-metadata)
+
+Follow these steps to create the module and configure its metadata for React:
+
+1.  [Create an OSGi module](/develop/tutorials/-/knowledge_base/7-1/starting-module-development#creating-a-module). 
+    For example, use the 
+    [npm React portlet template](/develop/reference/-/knowledge_base/7-1/npm-react-portlet-template). 
+
+2.  Specify the `Web-ContextPath` BND Header in your project's `bnd.bnd` file. 
+    Below is the default configuration for the npm React portlet template:
+
+        Web-ContextPath: /my-npm-react-portlet
+
+3.  Create a 
+    [`.babelrc` file](/develop/reference/-/knowledge_base/7-1/configuring-liferay-npm-bundler) 
+    and add the following presets to it:
+
+        {
+        	"presets": ["es2015", "react", "liferay-project"]
+        }
+
+4.  Add a 
+    [`.npmbundlerrc` file](/develop/reference/-/knowledge_base/7-1/configuring-liferay-npm-bundler) 
+    to your project's root folder and add the 
+    [`liferay-npm-bundler-preset-react` preset](https://www.npmjs.com/package/liferay-npm-bundler-preset-react). 
+    This preset provides the required `.npmbundlerrc` configuration. You can, 
+    however, copy the contents of the preset and add additional configuration 
+    options to the file instead if needed:
+
+        {
+            "preset": "liferay-npm-bundler-preset-react"
+        }
+
+5.  Include the following dependency to your `build.gradle` file:
+
+        compileOnly group: "com.liferay", 
+        name: "com.liferay.frontend.js.loader.modules.extender.api", 
+        version: "2.0.2"
+
+6.  Create a `package.json` in your project if it doesn't already exist and add 
+    the configuration shown below to it. Update the `"main"` JS path to reflect 
+    your app's main JS file (remember to add the `.es.js` extension if your file 
+    uses it). Note that the `liferay-npm-bundler` is added last to the build 
+    script. List any additional build processes before this that your project 
+    requires:
+    
+        {
+        	"dependencies": {
+        		"react": "15.6.2",
+        		"react-dom": "15.6.2"
+        	},
+        	"devDependencies": {
+        		"babel-cli": "6.26.0",
+        		"babel-preset-es2015": "6.24.1",
+        		"babel-preset-liferay-project": "1.6.1",
+        		"babel-preset-react": "6.24.1",
+        		"liferay-npm-bundler": "1.6.1",
```

Comment:
Version series `1.x` are already discouraged. I don't mind having a tutorial for `1.x` in case someone wants to keep using it, but I would clearly state the fact that `1.x` is discouraged at the beginning of the tutorial.

Reply:
Let's simply not document or show usages of bundler `1.x.x` and focus on `2.x.x` everywhere 👍

Reply:
Version 1.6.1 was automatically generated using the latest project templates. I'm pinging Cody to let him know that the templates need updated to use version 2.x (2.2.0 at the time of this writing)

Reply:
Yeah, you might need to specify you want them generated for `7.1` or something. @izaera, do we have the updated samples somewhere for Mike to use them?

Reply:
@jbalsas @izaera I'm guessing the `babel-preset-liferay-project` and the `liferay-npm-bundler-preset-react` need updated to version 2.2.0 as well

Reply:
Nope. There's no need to use presets at all. See the examples in https://github.com/liferay/liferay-blade-samples/tree/7.1/gradle/apps/npm

Reply:
Bundler 2.x doesn't need presets and it needs also changes to the .babelrc file to remove the `liferay-project` preset, for example.

Reply:
Right. We still need to update the project templates to use version 2.x.x to match those gradle sample configurations. Thanks @izaera ! I'll update the docs for now and let Cody know to use the samples as a guide.